block/manager.go

```diff
@@ -351,26 +351,27 @@ func (m *Manager) sendNonBlockingSignalToRetrieveCh() {
 	}
 }
 
-// trySyncNextBlock tries to progress one step (one block) in sync process.
+// trySyncNextBlock tries to execute as many blocks as possible.
 //
-// To be able to apply block and height h, we need to have its Commit. It is contained in block at height h+1.
-// If block at height h+1 is not available, value of last gossiped commit is checked.
+// For every block, to be able to apply block at height h, we need to have its Commit. It is contained in block at height h+1.
 // If commit for block h is available, we proceed with sync process, and remove synced block from sync cache.
 func (m *Manager) trySyncNextBlock(ctx context.Context, daHeight uint64) error {
-	var commit *types.Commit
-	currentHeight := m.store.Height() // TODO(tzdybal): maybe store a copy in memory
-
-	b, ok := m.blockCache.getBlock(currentHeight + 1)
-	if !ok {
-		return nil
-	}
+	for {
+		currentHeight := m.store.Height() // TODO(tzdybal): maybe store a copy in memory
```

Comment:
Let's either address this TODO or delete the comment. This is a pretty old TODO so if it is no longer relevant we should just delete.